sycl/include/sycl/ext/oneapi/experimental/root_group.hpp

```diff
@@ -0,0 +1,134 @@
+//==--- root_group.hpp --- SYCL extension for root groups ------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <sycl/builtins.hpp>
+#include <sycl/ext/oneapi/properties/properties.hpp>
+#include <sycl/memory_enums.hpp>
+#include <sycl/queue.hpp>
+
+#define SYCL_EXT_ONEAPI_ROOT_GROUP 1
+
+namespace sycl {
+__SYCL_INLINE_VER_NAMESPACE(_V1) {
+namespace ext::oneapi::experimental {
+
+namespace info::kernel_queue_specific {
+struct max_num_work_group_sync {
+  using return_type = size_t;
+};
+} // namespace info::kernel_queue_specific
+
+struct use_root_sync_key {
+  using value_t = property_value<use_root_sync_key>;
+};
+
+inline constexpr use_root_sync_key::value_t use_root_sync;
```

Comment:
The extension's [implementation notes](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_root_group.asciidoc#implementation-notes) mention the need for this property to cause special launches. Is this part of the TODOs too?

Reply:
Alright. As long as that was intended to be part of the initial implementation, I will disregard that part.